diff -Naur a/security/sandbox/linux/moz.build b/security/sandbox/linux/moz.build
--- a/security/sandbox/linux/moz.build	2016-06-01 12:11:46.000000000 +0800
+++ b/security/sandbox/linux/moz.build	2016-06-30 13:38:52.561530457 +0800
@@ -79,7 +79,7 @@
 # from the function using it which breaks the build.  Work around that by
 # forcing there to be only one partition.
 if '-flto' in CONFIG['OS_CXXFLAGS'] and not CONFIG['CLANG_CXX']:
-    LDFLAGS += ['--param lto-partitions=1']
+    LDFLAGS += ['--lto-partition=one']
 
 DEFINES['NS_NO_XPCOM'] = True
 DISABLE_STL_WRAPPING = True

