From 8f85c44954826519267f0c4bdf59771bfe80f511 Mon Sep 17 00:00:00 2001
From: Shane Peelar <lookatyouhacker@gmail.com>
Date: Thu, 16 Aug 2018 18:01:18 -0400
Subject: [PATCH] Fix bash_cv_must_reinstall_sighandlers check

---
 aclocal.m4 | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/aclocal.m4 b/aclocal.m4
index d3bac07..8462326 100644
--- a/aclocal.m4
+++ b/aclocal.m4
@@ -1307,7 +1307,7 @@ AC_CACHE_VAL(bash_cv_must_reinstall_sighandlers,
 
 typedef RETSIGTYPE sigfunc();
 
-int nsigint;
+volatile sig_atomic_t nsigint;
 
 #ifdef HAVE_POSIX_SIGNALS
 sigfunc *
-- 
2.18.0

